api/src/main/java/com/github/retrooper/packetevents/protocol/npc/NPC.java

```diff
@@ -270,51 +313,53 @@ public void updateNameTag(NPC npc) {
     }
 
     public void updateEquipment() {
-        for (Object channel : channels) {
-            List<Equipment> equipmentList = new ArrayList<>();
-            ItemStack handItem = getMainHand();
-            if (handItem == null) {
-                handItem = ItemStack.EMPTY;
-            }
-            equipmentList.add(new Equipment(EquipmentSlot.MAIN_HAND,
-                    handItem));
-            if (PacketEvents.getAPI().getServerManager().getVersion().isNewerThanOrEquals(ServerVersion.V_1_9)) {
-                ItemStack offHandItem = getOffHand();
-                if (offHandItem == null) {
-                    offHandItem = ItemStack.EMPTY;
-                }
-                equipmentList.add(new Equipment(EquipmentSlot.OFF_HAND,
-                        offHandItem));
-            }
-            ItemStack helmetItem = getHelmet();
-            if (helmetItem == null) {
-                helmetItem = ItemStack.EMPTY;
-            }
-            equipmentList.add(new Equipment(EquipmentSlot.HELMET,
-                    helmetItem));
+        if (channels.isEmpty()) return;
 
-            ItemStack chestPlateItem = getChestplate();
-            if (chestPlateItem == null) {
-                chestPlateItem = ItemStack.EMPTY;
-            }
-            equipmentList.add(new Equipment(EquipmentSlot.CHEST_PLATE,
-                    chestPlateItem));
-            ItemStack leggingsItem = getLeggings();
-            if (leggingsItem == null) {
-                leggingsItem = ItemStack.EMPTY;
-            }
-            equipmentList.add(new Equipment(EquipmentSlot.LEGGINGS,
-                    leggingsItem));
-            ItemStack bootsItem = getBoots();
-            if (bootsItem == null) {
-                bootsItem = ItemStack.EMPTY;
+        List<Equipment> equipmentList = new ArrayList<>();
+        ItemStack handItem = getMainHand();
+        if (handItem == null) {
+            handItem = ItemStack.EMPTY;
+        }
+        equipmentList.add(new Equipment(EquipmentSlot.MAIN_HAND,
+                handItem));
+        if (PacketEvents.getAPI().getServerManager().getVersion().isNewerThanOrEquals(ServerVersion.V_1_9)) {
+            ItemStack offHandItem = getOffHand();
+            if (offHandItem == null) {
+                offHandItem = ItemStack.EMPTY;
             }
-            equipmentList.add(new Equipment(EquipmentSlot.BOOTS,
-                    bootsItem));
+            equipmentList.add(new Equipment(EquipmentSlot.OFF_HAND,
+                    offHandItem));
+        }
+        ItemStack helmetItem = getHelmet();
+        if (helmetItem == null) {
+            helmetItem = ItemStack.EMPTY;
+        }
+        equipmentList.add(new Equipment(EquipmentSlot.HELMET,
```

Comment:
And why do you put "helmetItem" in another line, put them together.
(Please everywhere)
```suggestion
        equipmentList.add(new Equipment(EquipmentSlot.HELMET, helmetItem));
```

Reply:
You basically refactor the code at least for a little bit if you are going to change so much